### PR TITLE
Support for python 3.10 in ML extension

### PR DESCRIPTION
**Pull request by harneetvirk**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ML

**Review by wangzelin007 — APPROVED** [review]

File: src/index.json [hunk 1]
```diff
@@ -23619,6 +23619,73 @@
...
+                                "pathspec (==0.9.*)",
+                                "pydash (<=4.9.0)",
+                                "pyjwt (<=2.3.0)",
+                                "tqdm (<=4.63.0)",
+                                "typing-extensions (<4.0.0,>=3.10)"
+                            ]
+                        }
+                    ],
+                    "summary": "Microsoft Azure Command-Line Tools AzureMachineLearningWorkspaces Extension",
+                    "version": "2.2.2"
```
Comment (wangzelin007) [inline src/index.json:23686]:
```suggestion
                    "version": "2.2.3"
```

Comment (harneetvirk) [reply src/index.json:23686]: Updated the version in the summary

File: src/index.json [hunk 2]
```diff
@@ -23619,6 +23619,73 @@
...
+                                "pyjwt (<=2.3.0)",
+                                "tqdm (<=4.63.0)",
+                                "typing-extensions (<4.0.0,>=3.10)"
+                            ]
+                        }
+                    ],
+                    "summary": "Microsoft Azure Command-Line Tools AzureMachineLearningWorkspaces Extension",
+                    "version": "2.2.3"
+                },
+                "sha256Digest": "6CBFAC0985961D82AE22FE7BCCAC9CEE8A26C78958756F463CED58913EF96409"
```
Comment (wangzelin007) [inline src/index.json:23688]:
```suggestion
                "sha256Digest": "6cbfac0985961d82ae22fe7bccac9cee8a26c78958756f463ced58913ef96409"
```

**Review by wangzelin007 — CHANGES_REQUESTED** [review]
Please use all lowercase format.
Otherwise CI will report an error.
image

File: src/index.json (hunk 2 above)
Comment (harneetvirk) [reply src/index.json:23688]: Already pushed the change for hash in lower case.

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**